app/src/main/java/com/grzegorzdec/tagmango/seller/map/MapViewBindings.kt

```diff
@@ -0,0 +1,50 @@
+package com.grzegorzdec.tagmango.seller.map
+
+import androidx.databinding.BindingAdapter
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.MapView
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.LatLngBounds
+import com.google.android.gms.maps.model.MarkerOptions
+import com.grzegorzdec.tagmango.model.Client
+
+@BindingAdapter("mapMarkers")
+fun mapMarkers(mapView: MapView, clients: List<Client>) {
+    if(clients.isNotEmpty()) {
+        mapView.getMapAsync { googleMap ->
+            val latLngBuilder = LatLngBounds.Builder()
+            clients.forEach {
+                val latLng = LatLng(it.latitude, it.longitude)
+                latLngBuilder.include(latLng)
+                googleMap.addMarker(
+                    MarkerOptions()
+                        .position(latLng)
+                        .title(it.name)
+                )
+            }
+            val latLngBounds = latLngBuilder.build()
+            googleMap.animateCamera(CameraUpdateFactory.newLatLngBounds(latLngBounds, 120), 300, null)
+        }
+    }
```

Comment:
add `mapview.onCreate()` there

Reply:
But mapView onCreate should be invoked in onCreateView I think. This binding is triggered after list of the clients is downloaded from the API. So there is a possibility that this method would not be triggered at all

Reply:
load emptylist at the beggining